classes/war/WarAction.php

```diff
@@ -328,106 +317,133 @@ public function progressActiveOperation(): string
+                    $this->system->db->query("UPDATE `region_locations` SET `stability` = {$location_target['stability']} WHERE `region_location_id` = {$this->target_id}");
+                } else {
+                    $message .= "\nTarget Stability already at " . $max_stability . "!";
+                }
                 break;
-            case self::OPERATION_RAID:
+            case self::WAR_ACTION_RAID:
                 WarLogManager::logAction($this->system, $this->user, 1, WarLogManager::WAR_LOG_RAID, $this->target_village);
                 $defense_reduction = 1 + $this->user->village->policy->raid_defense;
+                $stability_reduction = 1 + $this->user->village->policy->raid_stability;
                 if ($location_target['health'] <= 0 && $location_target['type'] == 'castle') {
                     WarLogManager::logRegionCapture($this->system, $this->user, $location_target['region_id']);
                     // change region ownership
                     $this->system->db->query("UPDATE `regions` SET `village` = {$this->user->village->village_id} WHERE `region_id` = {$location_target['region_id']}");
-                    // update castle health to 50%, defense to 25
-                    $castle_hp = floor(WarManager::BASE_CASTLE_HEALTH * 0.5);
-                    $this->system->db->query("UPDATE `region_locations` SET `health` = {$castle_hp}, `defense` = 25 WHERE `region_id` = {$location_target['region_id']} AND `type` = 'castle'");
+                    // update castle health to 50%, defense to 25, stability to 25, occupying village
+                    $castle_hp = floor(WarManager::BASE_CASTLE_HEALTH * (WarManager::INITIAL_LOCATION_CAPTURE_HEALTH_PERCENT / 100));
+                    $castle_defense = WarManager::INITIAL_LOCATION_CAPTURE_DEFENSE;
+                    $castle_stability = WarManager::INITIAL_LOCATION_CAPTURE_STABILITY;
+                    $this->system->db->query("UPDATE `region_locations` SET `health` = {$castle_hp}, `defense` = {$castle_defense}, `stability` = {$castle_stability}, `occupying_village_id` = {$this->user->village->village_id} WHERE `region_id` = {$location_target['region_id']} AND `type` = 'castle'");
                     // update patrols, move back 5 minutes
                     $patrol_spawn = time() + (60 * 5);
                     $this->system->db->query("UPDATE `patrols` SET `start_time` = {$patrol_spawn}, `village_id` = {$this->user->village->village_id} WHERE `region_id` = {$location_target['region_id']}");
                     // update caravans, change only caravans that haven't spawned
                     $name = VillageManager::VILLAGE_NAMES[$this->user->village->village_id] . " Caravan";
                     $time = time();
                     $this->system->db->query("UPDATE `caravans` SET `village_id` = {$this->user->village->village_id}, `name` = '{$name}' WHERE `region_id` = {$location_target['region_id']} AND `start_time` > {$time}");
```

Comment:
These blocks of the switch statement are definitely long enough to need their own methods